db_controller.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from sqlite3 import Error
+import sqlite3
+
+
+def create_connection(db_file):
+    """ create a database connection to a SQLite database """
+    conn = None
+    try:
+        conn = sqlite3.connect(db_file)
+    except Error as e:
+        logging.getLogger('DB create').error(e)
```

Comment:
This works, but there's a cleaner way to do logging. create a global logger for each module, just after your imports
```python
LOG = getLogger(__name__)
```
`__name__` is a magic variable that contains the module name
Then, just `LOG.error(e)`. This lets you then set filters / handlers for this one module's logger uniformly.

Reply:
This applies to all other occurrences of `getLogger()` in the other modules as well (I won't spam you with these comments)